File: js/src/types/attestation-common.ts
/** Signing algorithms exposed by NEAR AI Cloud. */
export type SigningAlgorithm = 'ecdsa' | 'ed25519';

/** Public identity of the key that signs a completion or attestation. */
export type SigningIdentity = {
  algorithm: SigningAlgorithm;
  address: string;
};

/** dstack event log as returned by the attestation endpoint. */
export type AttestationEventLog = string | readonly unknown[];

/** Shared raw evidence consumed by model and gateway verification. */
export type AttestationEvidence = {
  nonce: string;
  signer: SigningIdentity;
  intelQuote: string;
  eventLog: AttestationEventLog;
  appCompose: string;
  declaredSpkiFingerprint?: string | null;
  reportedQuoteData?: string;
};

File: py/src/verifiable_ai_sdk/types/attestation_common.py
"""Shared public attestation data types."""

from __future__ import annotations

from dataclasses import dataclass
from typing import Literal, get_args


SigningAlgo = Literal['ecdsa', 'ed25519']
TcbStatus = Literal[
    'UpToDate',
    'SWHardeningNeeded',
    'ConfigurationNeeded',
    'ConfigurationAndSWHardeningNeeded',
    'OutOfDate',
    'OutOfDateConfigurationNeeded',
    'Revoked',
    'Unknown',
]
SUPPORTED_TCB_STATUSES: frozenset[TcbStatus] = frozenset(get_args(TcbStatus))
AttestationEventLog = str | list[object]


@dataclass(frozen=True, kw_only=True)
class SigningIdentity:
    """The public key identity advertised by evidence or a signature."""

    signing_algo: SigningAlgo
    signing_address: str


@dataclass(frozen=True, kw_only=True)
class AttestationEvidence:
    """Fields shared by raw model and Gateway evidence."""

    nonce: str
    signer: SigningIdentity
    intel_quote: str
    event_log: AttestationEventLog
    app_compose: str
    declared_spki_fingerprint: str | None = None
